src/aap_eda/conf/registry.py

```diff
@@ -140,14 +140,21 @@ def db_update_setting(self, key: str, value: Any) -> None:
         Setting.objects.filter(key=key).update(
             value=self._setting_value(key, value)
         )
+        if key == "AUTOMATION_ANALYTICS_GATHER_INTERVAL":
+            SettingsRegistry._reschedule_gather_analytics(int(value))
 
     def db_update_settings(self, settings: dict[str, Any]) -> None:
+        interval = None
         with transaction.atomic():
             for key, value in settings.items():
                 self._validate_key(key, writable=True)
                 Setting.objects.filter(key=key).update(
                     value=self._setting_value(key, value)
                 )
+                if key == "AUTOMATION_ANALYTICS_GATHER_INTERVAL":
+                    interval = int(value)
+        if interval:
+            SettingsRegistry._reschedule_gather_analytics(interval)
```

Comment:
We could save a little duplicate code by re-writing `db_update_setting` as a call to `db_update_settings({key: value})`